neuro_san/internals/graph/activations/calling_activation.py

```diff
@@ -222,9 +222,12 @@ async def make_one_tool_function_call(self, component_tool_call: ToolCall) -> Di
         callable_component: CallableActivation = \
             self.factory.create_agent_activation(self.run_context, our_agent_spec, use_tool_name,
                                                  self.sly_data, tool_arguments)
-        callable_component_response: List[Any] = await callable_component.build()
 
-        output: str = json.dumps(callable_component_response)
+        output: str = await callable_component.build()
+        # Even though we get a string, run it through the json stuff again to more reliably
+        # escape when the output itself has JSON in it.  When messing with this, it's worth
+        # testing both esp_decision_assistant and intranet_agents_with_tools.
+        output = json.dumps(output)
```

Comment:
You would think that if we were already getting a string from build(), we would not need to pass it through json.dumps() to convert it to a string again.  Well, it turns out you'd be mistaken.

In order for the chicken scenario to be successful here, the double -> single flip above needs to be in place.
That's all fine, but if this extra json.dumps() is taken away, then the JSON in intranet_agents_with_tools goes ahoo. 🤷